ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -553,140 +547,176 @@ let definition_query (state : State.t) uri position merlin_request =
   let open Fiber.O in
   let+ result = Document.dispatch_exn doc command in
   let result = location_of_merlin_loc uri result in
-  Ok (result, state)
+  Ok result
+
+let highlight (state : State.t)
```

Comment:
Might be nice to have the same pattern for all of the requests, e.g. extract `prepare_rename`. But outside of the scope of this PR.